12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
 
-The concept of too small for `dispute_delta` is also subjective. The `dispute_delta` defines how many blocks the server will have to respond with the `justice_transaction`. The smallest the value, the more the server risks to fail the appointment.
+If `accountability` is not being offered, it makes not much sense accepting appointments that request it. If the tower accepts an appointment requesting `accountability`, it should be enforced or refunded. Generally, this is trying to allow a reputationally accountable watching service. The signed job from the customer provides an explicit acknowledgement of the transaction details that is important for the WatchTower to decide whether it can accept them. If the decrypted justice transaction does not satisfy the signed job (e.g. fee too low), then the WatchTower is not obliged to fulfil the appointment. 
 
-`encrypted_blob` should have been encrypted using `cipher`. Block ciphers have a size multiple of the block length, which depends on the key size. Therefore some incorrect `encrypted_blob` can be spotted checking the `transaction_size`. Moreover, `encrypted_blob` have to be at least as big as:
+#### `tower_evidence`
 
-`ceil(minimum_viable_transaction_size / cipher_block_size)`
+The format for the `tower_evidence` is defined as follows:
 
-And at most as big as:
+1. type: ? (`tower_evidence`)
+2. data:  
+	* [`u16 `:`receipt_length`]
+	* [`receipt_length `: `receipt`]
+	* [`u16`: `wt_signature_algorithm`]
+	* [`u16`: `wt_signature_len`
+	* [`wt_signature_len*byte`: `wt_signature`]
+	* [`u16`: `wt_public_key_len`]
+	* [`wt_public_key_len*byte`: `wt_public_key`]
 
-`ceil(maximum_viable_transaction_size / cipher_block_size`) 
+The sending node:
 
-`minimum_viable_transaction_size` and `maximum_viable_transaction_size` refer to the minimum/maximum size required to create a valid transaction.
+* MUST set `receipt` to a receipt built according to 	[Receipt-Format](#receipt-format)
+* MUST set `wt_signature_algorithm` to one of the signature algorithms he has announced.
+* MUST set `wt_signature` to the signature of the appointment using `wt_signature_algorithm`.
+* MUST set `wt_public_key` to the public key that matches the private key used to create `wt_signature`.
 
-Accepting `encrypted_blob` outside those boundaries will ease DoS attackes on the server.
- 
+The receiving node:
+
+* MUST compute the `wt_signature` verification using `wt_public_key`.
+
+* MUST fail the connection if:
+	* Any of the fields is missing.
+	* `receipt` does not matches the format specified at 	[Receipt-Format](#receipt-format)
+	* `receipt` fields do not match the ones sent in the `appointment` message.
+	* `wt_signature_algorithm` does not match any of the ones offered by the WatchTower
+	* `wt_signature` cannot be verified using `wt_public_key`.
+
+#### Receipt Format 
+
+The server (WatchTower) MUST create the receipt as containing the following infiormation:
```

Comment:
Some places you call this `server`, others `WatchTower`. Maybe it would be best if you say somewhere at the very start something like: "For this rest of this document wherever we will refer to the `WatchTower` as `server`, and the customer as `client`."